File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cs
using System;

public class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        int[] arr = new int[number + 1];
        
        for(int i = 1; i <= number; i++)
        {
            int cnt = 0;
            int r = (int)Math.Sqrt(i);
            for(int j = 1; j <= r; j++)
            {
                if(i % j == 0)
                {
                    int k = i / j;
                    cnt += (j == k) ? 1 : 2;
                }
            }
            arr[i] = cnt;
        }
        foreach(int i in arr)
        {
            if(i > limit) answer += power;
            else answer += i;
        }
        return answer;
    }
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>

using namespace std;

int check(int num)
{
    int cnt = 0;
    for(int i = 1; i * i <= num; i++)
    {
        if(num % i == 0)
        {
            if(i * i == num)
                cnt++;
            else
                cnt += 2;
        }
    }
    return cnt;
}

int solution(int number, int limit, int power) {
    int answer = 0;
    vector<int> v;
    for(int i = 1; i <= number; i++)
    {
        v.push_back(check(i));
    }
    
    for(int i : v)
    {
        if(i > limit)
            answer += power;
        else
            answer += i;
    }
    return answer;
}